grunnbelop/src/test/kotlin/no/nav/dagpenger/grunnbelop/GrunnbelopTest.kt

```diff
@@ -8,67 +14,118 @@ import java.time.Month
+        )
     }
 
     @Test
     fun ` Skal returnere grunnbeløp på 92576 for måned mai 2016  `() {
-
-        Assertions.assertEquals(92576.toBigDecimal(), getGrunnbeløpForMåned(YearMonth.of(2016, Month.MAY)).verdi)
+        Assertions.assertEquals(
+            92576.toBigDecimal(),
+            getGrunnbeløpForRegel(Regel.Grunnlag).forMåned(YearMonth.of(2016, Month.MAY)).grunnbeløp.verdi
+        )
     }
 
     @Test
     fun ` Skal returnere grunnbeløp på 90068 for måned august 2015 `() {
-
-        Assertions.assertEquals(90068.toBigDecimal(), getGrunnbeløpForMåned(YearMonth.of(2015, Month.AUGUST)).verdi)
+        Assertions.assertEquals(
+            90068.toBigDecimal(),
+            getGrunnbeløpForRegel(Regel.Grunnlag).forMåned(YearMonth.of(2015, Month.AUGUST)).grunnbeløp.verdi
+        )
     }
 
     @Test
     fun ` Skal returnere grunnbeløp på 90068 for dato 06082015 `() {
-
-        Assertions.assertEquals(90068.toBigDecimal(), getGrunnbeløpForDato(LocalDate.of(2015, 8, 6)).verdi)
+        Assertions.assertEquals(
+            90068.toBigDecimal(),
+            getGrunnbeløpForRegel(Regel.Grunnlag).forDato(LocalDate.of(2015, 8, 6)).grunnbeløp.verdi
+        )
     }
 
     @Test
     fun ` Skal returnere grunnbeløp på 90068 for dato 01052019 `() {
-
-        Assertions.assertEquals(99858.toBigDecimal(), getGrunnbeløpForDato(LocalDate.of(2019, 5, 1)).verdi)
+        Assertions.assertEquals(
+            99858.toBigDecimal(),
+            getGrunnbeløpForRegel(Regel.Grunnlag).forDato(LocalDate.of(2019, 5, 1)).grunnbeløp.verdi
+        )
     }
 
     @Test
     fun `Skal finne faktoren mellom to Grunnbeløp med desimaler`() {
+        val grunnbeløp = Grunnbeløp.FastsattI2017
+        val gjeldendeGrunnbeløp = Grunnbeløp.FastsattI2018
+
+        val faktor = gjeldendeGrunnbeløp.faktorMellom(grunnbeløp)
+        assertSoftly {
+            faktor shouldBe BigDecimal("1.03469893414785227588")
+            faktor.scale() shouldBe 20
+        }
+    }
+
+    @Test
+    fun `Alle grunnbeløp har en mapping`() {
+        Grunnbeløp.values().forAll { grunnbeløp ->
+            gyldighetsperioder shouldContainKey grunnbeløp
+        }
+    }
 
-        val grunnbeløp = Grunnbeløp(LocalDate.now(), LocalDate.now(), 93634.toBigDecimal())
-        val gjeldendeGrunnbeløp = Grunnbeløp(LocalDate.now(), LocalDate.now(), 96883.toBigDecimal())
+    @Test
+    fun `Alle regler har en mapping for hvert grunnbeløp`() {
+        Regel.values().forAll {regel ->
+            gyldighetsperioder.values.forAll {
+                it shouldContainKey regel
+            }
+        }
+    }
 
-        Assertions.assertEquals(BigDecimal("1.03469893414785227588"), gjeldendeGrunnbeløp.faktorMellom(grunnbeløp))
+    @Test
+    // This test is a bit awkward, but it's a nice way to blackbox verify that all mappings exists
+    fun `Alle kombinasjoner av år og regler har en mapping`() {
```

Comment:
måtte lese den noen ganger, men 👍

Reply:
Det er litt 💩 at testen er såpass strenge på at alt må være definert siden det gjør det vanskelig å innføre ny G for minsteinntekt uten å måtte styre det med deploy.

Reply:
Fordi ny G for grunnbeløp skal etter en eller annen magisk dato (f.eks. 18. Mai) tre i kraft fra 1. Mai.

Mulig vi kan utvide mappingen med en fom-dato til?